lib/dmv.rb

```diff
@@ -9,8 +13,30 @@ def add_facility(facility)
   end
 
   def facilities_offering_service(service)
-    @facilities.find do |facility|
+    @facilities.find_all do |facility|
       facility.services.include?(service)
     end
   end
+
+  def most_popular_make_model
+    return if @facilities.map(&:registered_vehicles).empty?
+
+    @facilities.map(&:registered_vehicles).flatten.reject { |vehicle| vehicle.model.nil? }.group_by do |vehicle|
+      "#{vehicle.make} #{vehicle.model}"
+    end.max_by { |_make_model, vehicle_array| vehicle_array.length }[0]
+  end
+
+  def count_model_year(year)
+    return if @facilities.map(&:registered_vehicles).empty?
+
+    @facilities.map(&:registered_vehicles).flatten.count { |vehicle| vehicle.year == year }
+  end
+
+  def most_popular_county
+    return if @facilities.map(&:registered_vehicles).empty?
+
+    @facilities.map(&:registered_vehicles).flatten.group_by(&:county).max_by do |_county, vehicle_array|
+      vehicle_array.length
+    end[0]
```

Comment:
Nice use of chained enumerables! As a complimentary exercise, it's a good idea to practice talking through these solutions as they are more complex, and describing your thought process and approach to a solution can be great interview practice!